scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -221,17 +222,32 @@ class FossId(
                         .checkResponse("create project")
                 }
 
-                log.info { "Creating scan for URL $url and revision $revision ..." }
+                val scans = service.listScansForProject(user, apiKey, projectCode)
+                    .checkResponse("list scans for project").data
+                checkNotNull(scans)
 
-                val scanCode = createScan(projectCode, url, revision)
+                val existingScan = scans.sortedByDescending { it.id }.find { scan ->
+                    scan.gitBranch == revision && scan.gitRepoUrl == url
+                }
 
-                log.info { "Initiating download from Git repository ..." }
+                val scanCode = if (existingScan == null) {
+                    log.info { "No scan found for $url and revision $revision. Creating scan ..." }
+                    val scanCode = createScan(projectCode, url, revision)
 
-                service.downloadFromGit(user, apiKey, scanCode)
-                    .checkResponse("download data from Git", false)
+                    log.info { "Initiating data download ..." }
+                    val responseDownload = service.downloadFromGit(user, apiKey, scanCode)
+                    responseDownload.checkResponse("download data from Git", false)
```

Comment:
As `responseDownload` is used nowhere else, maybe just call `checkResponse()` directly on `downloadFromGit()`?